### PR TITLE
Denis Bereza - L1

### PR DESCRIPTION
**Pull request by spearfisher**

### CODE REVIEW

File: lesson1.rb [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
+      'Invalid Date Format'
+    end
   end
 
   def name
-    # TODO
+    full_name = ''
+    3.times do |input|
+      input = gets
+      full_name = full_name + ' ' + input
```
Comment (Alexey-Lukin) [inline lesson1.rb]: `full_name += " #{input}"`

File: lesson1.rb [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+    end
+    'Hello' + full_name + '!'
```
Comment (Alexey-Lukin) [inline lesson1.rb]: Always prefer to use string interpolation instead of string concatenation:
`"Hello#{full_name}!"`

File: lesson1.rb [hunk 3]
```diff
@@ -1,14 +1,27 @@
...
 
   def age(birthday)
-    # TODO
+    if birthday
+      days = (Date.parse(Time.now.to_s) - Date.parse(birthday.to_s)).to_i
+      "Я живу #{days / 365} года или #{days} дней или #{days * 24} " \
+      "часов или #{days * 24 * 60} минут или #{days * 24 * 60 * 60} секунд"
+    else
+      'Invalid Date Format'
+    end
```
Comment (Alexey-Lukin) [inline lesson1.rb]:
```
def age(birthday)
  return 'Invalid Date Format' unless birthday
  days = (Date.parse(Time.now.to_s) - Date.parse(birthday.to_s)).to_i
  "Я живу #{days / 365} года или #{days} дней или #{days * 24} " \
  "часов или #{days * 24 * 60} минут или #{days * 24 * 60 * 60} секунд"
end
```

File: my_array.rb [hunk 4]
```diff
@@ -6,58 +6,66 @@ def initialize(arr = [])
...
-    # TODO
+    @array.map { |element| element.to_f / 10 }
   end
 
   def chars
-    # TODO
+    res = []
+    array.each { |element| res << ('a'..'z').to_a[element - 1].to_sym }
+    res
   end
```
Comment (Alexey-Lukin) [inline my_array.rb]: Avoid using unnecessary local variables:
`array.map { |element| ('a'..'z').to_a[element - 1].to_sym }`

File: my_array.rb [hunk 5]
```diff
@@ -6,58 +6,66 @@ def initialize(arr = [])
...
 
   def before_min
-    # TODO
+    @array.take(@array.index(@array.min))
   end
 
   def three_smallest
-    # TODO
+    @array.sort.take(3)
```
Comment (Alexey-Lukin) [inline my_array.rb]: Be consistent: some methods from your class work with `attr_accessor :array` and other methods work with `@array`.

**Comment by spearfisher** [comment]
Ok. Thanks for code review.

File: my_array.rb [hunk 6]
```diff
@@ -6,58 +6,66 @@ def initialize(arr = [])
...
-    # TODO
+    res = []
+    array.each { |element| res << ('a'..'z').to_a[element - 1].to_sym }
+    res
   end
 
   def switch
-    # TODO
+    min = array.min
+    max = array.max
```
Comment (Alexey-Lukin) [inline my_array.rb]: `min, max = array.minmax`

**Comment by spearfisher** [comment]
Interesting, when i used multiple assignment with this code: 
 array[max], array[min] = array[min], array[max]
rubocop said: Do not to use parallel assignment.

But that code, passed the test:
 min, max = @array.minmax
@array[max_index], @array[min_index] = @array.minmax

Why we should not use parallel assignment?

**Comment by Alexey-Lukin** [comment]
@spearfisher, according to https://github.com/bbatsov/ruby-style-guide#parallel-assignment
`Parallel assignment is allowed when it is the return of a method call, used with the splat operator, or when used to swap variable assignment`

**Comment by spearfisher** [comment]
Excellent repo!!! Thank you/

But in my first version of code I used swap variable assignment. Maybe rubocop don't know about it.

 def switch
     max = array.index(array.min)
     min = array.index(array.max)
     array[max], array[min] = array[min], array[max]
     array
 end

**Comment by Alexey-Lukin** [comment]
@spearfisher, you could find here [rubocop](https://github.com/bbatsov/rubocop/blob/bf63c3e3dcf4eba5a8d26dff69616ce89020a22a/spec/rubocop/cop/style/parallel_assignment_spec.rb) what is allowed by rubocop and what will be identified as offence by rubocop

**Comment by spearfisher** [comment]
Ill find it:  
   it_behaves_like('allowed', 'a, b = b, a')
I should place array[max] and array[min] into a variables.

**Pull request closed without merging**